### DISCUSSION TITLE
Sioyek is amazing! I would love to be able to integrate it with latextools and sublimetext and to use sioyek to easily annotate PDFs and export annotations to a separate .txt or .md file

### DISCUSSION THREAD

**Discussion by gbstan** [ahrm/sioyek#553] [Ideas]
Hi there,
Thank you so much for developing such a powerful PDF viewer. It has features that are very useful and that I never seen before (one only realise how useful they are when we see them and use them).

I would love to be able to integrate sioyek with latextools and sublimetext, especially for synctex. I have tried to do so but couldn't really get it to work on a mac.

Another feature that would make me switch to sioyek from skim.app on the mac would be the ability to use sioyek to easily add text comments to a PDF, and then export these comments to a .txt or .md file. This feature of skim.app on a mac is really helpful to easily annotate a PDF and create a text report from text comments/annotations and highlights too (in skim you can highlight and add text to the highlighted text too).

**Comment by ahrm** [comment]
> I would love to be able to integrate sioyek with latextools and sublimetext, especially for synctex. I have tried to do so but couldn't really get it to work on a mac.

I don't use sublime text so I don't know the exact commands, but the synctex feature is completely configurable, so you should be able to configure it to work for any viewer. See [the documentation](https://sioyek-documentation.readthedocs.io/en/latest/usage.html#synctex) for more information. 

> ability to use sioyek to easily add text comments to a PDF, and then export these comments to a .txt or .md file

This could be added as an extensions. We already have an [extension](https://github.com/ahrm/sioyek-python-extensions#-add_text) that  adds text annotations. An extension that export these into a new document is pretty simple.

**Comment by gbstan** [comment]
Thank you for the quick response. I have tried to configure synctex to work with the sublimetext package latextools but couldn't get it to work.

Here is the best information I could find: https://latextools.readthedocs.io/en/latest/available-viewers/#preview and I tried to use this to get a latextools configuration in sublimetext to use sioyek but couldn't get it to work (not sure why).

For the annotation export to .txt or .md, I am not really versed into programming, so I have not tried myself.

I guess my suggestion is about some features that other pdf viewers/editors currently implement that might be useful to have with sioyek too.

**Comment by gbstan** [comment]
I think I got it to work.

The change for the configuration of latextools should look something like this:
```
"viewer": "command",
"viewer_settings": {
		// Platform-specific settings:
		"osx" : {
			"forward_sync_command": "sioyek --execute-command toggle_synctex --reuse-window $pdf_file --forward-search-file $src_file --forward-search-line $line",
                "view_command": "sioyek"
		},
}
```

**Comment by gbstan** [comment]
One more question: I am trying to get sioyek to execute two commands in the configuration above, but it seems only toggle_synctex  is taken into consideration:

sioyek --execute-command toggle_synctex --execute-command toggle_presenation_mode

or do I not have the right syntax to pass on two commands to sioyek?

**Reply by ahrm** [reply]
You can do it like this:
```
sioyek --execute-command toggle_synctex;toggle_presenation_mode
```

**Comment by gbstan** [comment]
Yes, that is what I thought too; however when I try this below in the settings of latextools, I get synctex but not presentation-mode:
```
"viewer_settings": {
		// Platform-specific settings:
		"osx" : {
			"forward_sync_command": "sioyek --execute-command toggle_synctex;toggle_presenation_mode --reuse-window $pdf_file --forward-search-file $src_file --forward-search-line $line",
                "view_command": "sioyek"
			// See README or third-party documentation
		},
}
```

**Comment by ahrm** [comment]
That's because you misspelled `presentation`.

**Reply by gbstan** [reply]
Yup, you are right! Apologies for this.
One last question. It seems that with the current setup, invoking "sioyek --execute-command toggle_synctex;toggle_presentation_mode --reuse-window $pdf_file --forward-search-file $src_file --forward-search-line $line" will toggle synctex and presentation mode ON -> OFF and OFF -> ON each time, which means that each time we compile latex to pdf or we do a forward search using such a command, synctex (and presentation mode) will toggle from ON to OFF or OFF to ON. Is there a way to integrate and invoke repeatedly sioyek with syntex (or presentation mode) staying active instead of toggling?

I know this could be done via the prefs_user file, but I would like to have these options only active when sioyek is invoked via a text editor for latex compilation and viewing of the pdf file.

**Reply by ahrm** [reply]
I just added this option today (for synctex mode, I could also add it for presentation mode). See https://github.com/ahrm/sioyek/issues/556.

**Comment by gbstan** [comment]
Amazing! I wonder if there are other options that would benefit from staying ON or OFF beyond those two for user-cases like this one.

**Reply by ahrm** [reply]
Added for presentation mode in b511a5846f088b6e6b24d6e2cf9763027f3e5430.

**Comment by gbstan** [comment]
For future reference here is the modification to sublimetext's latextools package for forward sync:

```
"viewer": "command",
"viewer_settings": {
		// Platform-specific settings:
		"osx" : {
			"forward_sync_command": "sioyek --execute-command turn_on_synctex;turn_on_presentation_mode --reuse-window $pdf_file --forward-search-file $src_file --forward-search-line $line",
                "view_command": "sioyek"
		},
},
```

**Reply by gbstan** [reply]
Thank so much for being responsive on these questions and requests. Looking forward to the next release of sioyek that integrates these changes
